lib/nanoc/base/entities/configuration.rb

```diff
@@ -47,6 +51,21 @@ def with_defaults
       self.class.new(new_wrapped)
     end
 
+    def with_environment(env = nil)
```

Comment:
`with_environment` does not appear to be called with an argument except for tests. Make the tests set up the environment, and remove this argument.